crates/client-api/src/routes/database.rs

```diff
@@ -54,20 +55,28 @@ pub struct CallParams {
     reducer: String,
 }
 
+#[derive(Deserialize)]
+pub struct CallQueryParams {
+    client_address: Option<Address>,
+}
+
 pub async fn call(
     State(worker_ctx): State<Arc<dyn WorkerCtx>>,
     auth: SpacetimeAuthHeader,
     Path(CallParams {
         name_or_address,
         reducer,
     }): Path<CallParams>,
+    Query(CallQueryParams { client_address }): Query<CallQueryParams>,
     ByteStringBody(body): ByteStringBody,
 ) -> axum::response::Result<impl IntoResponse> {
     let SpacetimeAuth {
         identity: caller_identity,
         creds: caller_identity_token,
     } = auth.get_or_create(&*worker_ctx).await?;
 
+    let client_address = client_address.unwrap_or_else(generate_random_address);
```

Comment:
Yeah, this seems reasonable. I think instead of using the dummy address, I'd rather make `ReducerContext` hold an `Option<Address>`, but that feels morally equivalent.

Reply:
With commit [21819de](https://github.com/clockworklabs/SpacetimeDB/pull/299/commits/21819de71572b1596f44e8a462faaa509e746308), we reserve the all-zeros `Address`. We still pass a 16-byte buffer across the WASM boundary, but the bindings interpret all-zeros as `None`. The HTTP `call` endpoint passes `None` when no address query parameter is included.